pkg/monitors/pingdomtransaction/pingdom-transaction-monitor.go

```diff
@@ -0,0 +1,294 @@
+package pingdomtransaction
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	pingdomNew "github.com/karlderkaefer/pingdom-golang-client/pkg/pingdom/openapi"
+	"github.com/karlderkaefer/pingdom-golang-client/pkg/pingdom/openapi/ptr"
+	endpointmonitorv1alpha1 "github.com/stakater/IngressMonitorController/v2/api/v1alpha1"
+	"github.com/stakater/IngressMonitorController/v2/pkg/config"
+	"github.com/stakater/IngressMonitorController/v2/pkg/models"
+	"github.com/stakater/IngressMonitorController/v2/pkg/util"
+)
+
+var log = logf.Log.WithName("pingdom-transaction")
+
+// PingdomTransactionMonitorService interfaces with MonitorService
+type PingdomTransactionMonitorService struct {
+	apiToken          string
+	url               string
+	alertContacts     string
+	alertIntegrations string
+	teamAlertContacts string
+	client            *pingdomNew.APIClient
+	context           context.Context
+}
+
+func (monitor *PingdomTransactionMonitorService) Equal(oldMonitor models.Monitor, newMonitor models.Monitor) bool {
+	// TODO: Retrieve oldMonitor config and compare it here
+	return false
+}
+
+func (service *PingdomTransactionMonitorService) Setup(p config.Provider) {
+	service.apiToken = p.ApiToken
+	service.url = p.ApiURL
+	service.alertContacts = p.AlertContacts
+	service.alertIntegrations = p.AlertIntegrations
+	service.teamAlertContacts = p.TeamAlertContacts
+
+	cfg := pingdomNew.NewConfiguration()
+	if service.apiToken == "" {
+		service.apiToken = os.Getenv("PINGDOM_API_TOKEN")
+	}
+	cfg.SetApiToken(service.apiToken)
+	service.client = pingdomNew.NewAPIClient(cfg)
+	service.context = context.Background()
+}
+
+func (service *PingdomTransactionMonitorService) GetByName(name string) (*models.Monitor, error) {
+	var match *models.Monitor
+
+	monitors := service.GetAll()
+	for _, mon := range monitors {
+		if mon.Name == name {
+			return &mon, nil
+		}
+	}
+
+	return match, fmt.Errorf("unable to locate monitor with name %v", name)
+}
+
+func (service *PingdomTransactionMonitorService) GetAll() []models.Monitor {
+	var monitors []models.Monitor
+	checks, _, err := service.client.TMSChecksAPI.GetAllChecks(service.context).Type_("script").Execute()
+	if err != nil {
+		log.Error(err, "Error getting all transaction checks")
+		return monitors
+	}
+
+	if checks == nil {
+		return monitors
+	}
+	for _, mon := range checks.GetChecks() {
+		newMon := models.Monitor{
+			URL:  service.GetUrlFromSteps(*mon.Id),
+			ID:   fmt.Sprintf("%v", *mon.Id),
+			Name: *mon.Name,
+		}
+		monitors = append(monitors, newMon)
+	}
+	return monitors
+}
+
+func (service *PingdomTransactionMonitorService) GetUrlFromSteps(id int64) string {
```

Comment:
not sure if this required. This was a attempt before I realized that the controller is not supporting more than one provider

Reply:
as far as i have seen, it supports atleast two monitors, with working on both simultaneously

Reply:
yes stakater will send out request to every provider that is defined. This is not required in my opinion, since I don't the alert duplication on different provider, but instead I want to use different provider for different alerts.